addons/mail/models/mail_thread.py

```diff
@@ -1511,7 +1506,7 @@ def _message_extract_payload_postprocess(self, message, body, attachments):
             node.getparent().remove(node)
         if postprocessed:
             body = etree.tostring(root, pretty_print=False, encoding='UTF-8')
-            body = pycompat.to_native(body)
+            body = body
```

Comment:
This should instead be changing the encoding of etree.tostring to `unicode`. In P3, `to_native` was *decoding* the bytes to an str.